src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Search.cs

```diff
@@ -0,0 +1,273 @@
+
+            const int OverloadCount = 1;
+            var overloads = new List<TexlStrings.StringGetter[]>(OverloadCount);
+
+            // Limit the signature length of params descriptions.
+            int argCount = arity > SignatureConstraint.RepeatTopLength ? SignatureConstraint.RepeatTopLength : arity;
+            var overload = new TexlStrings.StringGetter[argCount];
+            overload[0] = TexlStrings.SearchArg1;
+            overload[1] = TexlStrings.SearchArg2;
+            for (int iarg = 2; iarg < argCount; iarg++)
+            {
+                overload[iarg] = TexlStrings.SearchArg3;
+            }
+
+            overloads.Add(overload);
+            return new ReadOnlyCollection<TexlStrings.StringGetter[]>(overloads);
+        }
+
+        public override bool HasSuggestionsForParam(int argumentIndex)
+        {
+            Contracts.Assert(argumentIndex >= 0);
+
+            return argumentIndex >= 0;
+        }
+
+        public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+            Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
+
+            bool fValid = base.CheckTypes(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
+            if (!fValid)
+            {
+                return fValid;
+            }
+
+            Contracts.Assert(returnType.IsTable);
+
+            var argLen = args.Length;
+            returnType = argTypes[0];
+            DType sourceType = argTypes[0];
+
+            var supportColumnNamesAsIdentifiers = context.Features.SupportColumnNamesAsIdentifiers;
+            if (argTypes[1].Kind != DKind.String)
+            {
+                errors.EnsureError(DocumentErrorSeverity.Severe, args[1], TexlStrings.ErrStringExpected);
+                fValid = false;
+            }
+
+            // Check if table contains any searchable columns (column type is string).
+            if (!sourceType.GetAllNames(DPath.Root).Any(name => name.Type.Kind == DKind.String))
+            {
+                errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrSearchWrongTableType);
+                fValid = false;
+            }
+
+            for (var i = 2; i < argLen; i++)
+            {
+                var nameArg = args[i];
+                if (!base.TryGetColumnLogicalName(argTypes[0], supportColumnNamesAsIdentifiers, nameArg, errors, out var columnName))
+                {
+                    fValid = false;
+                    continue;
+                }
+
+                var columnType = sourceType.GetType(columnName);
```

Comment:
>var columnType = sourceType.GetType(columnName) [](http://example.com/codeflow?start=16&length=47)

There is an overload of TryGetLogicalColumnName that also returns the column type, so you can use that and avoid this call.

Reply:
thanks for pointing that out, updated.